alma/views/js/alma-product-insurance.js

```diff
@@ -206,92 +226,101 @@
+        function addInsuranceInput(inputName, value, form) {
+            let $elementInput = $(`#${inputName}`)
+            if (!$elementInput.length) {
+                const $input = $(`<input type="hidden" id="${inputName}" name="${inputName}" class="alma_insurance_input" value="${value}">`)
+                $(form).prepend($input);
             } else {
-                elementInput.setAttribute('value', value);
+                $elementInput.val(value);
             }
         }
 
+        // Remove our insurance hidden fields from the add to cart form
+        function removeInsuranceInputs() {
+            $('.alma_insurance_input').remove()
+        }
+
         function removeInsurance() {
+            // !! Global function provided by openInPageModal script
             resetInsurance();
             insuranceSelected = false;
-            removeInputInsurance();
-        }
-
-        function removeInputInsurance() {
-            let inputsInsurance = document.getElementById('add-to-cart-or-refresh').querySelectorAll('.alma_insurance_input');
-            inputsInsurance.forEach((input) => {
-                input.remove();
-            });
+            removeInsuranceInputs();
         }
 
         function addModalListenerToAddToCart() {
             if (AlmaInsurance.settings.isAddToCartPopupActivated === true && almaEligibilityAnswer) {
                 const addToCart = getAddToCartButton();
                 if (addToCart) {
                     // If we change the quantity the DOM is reloaded then we need to remove and add the listener again
-                    addToCart.removeEventListener("click", insuranceListener);
-                    addToCart.addEventListener("click", insuranceListener);
+                    addToCart.removeEventListener("click", addToCartListener);
+                    addToCart.addEventListener("click", addToCartListener);
                 }
             }
         }
 
-        function getAddToCartButton() {
-            let addToCart = document.querySelector('button.add-to-cart');
-            // TODO: Ravate specific to generalise with selector configuration
-            if (!addToCart) {
-                addToCart = document.querySelector('.add-to-cart a, .add-to-cart button').first();
-            }
+        // Find the selector for the add to cart button
+        function getAddToCartBtnSelector() {
+            // TODO: move selectors to a module configuration option
+            const selectors = [
+                '[data-button-action=add-to-cart]:visible', // generic PrestaShop add to cart button
+                'button.add-to-cart:visible',
+                '.add-to-cart a:visible',
+                '.add-to-cart button:visible',
+                'a[role=button][href$=addToCart]:visible' // elementor addToCart widget
+            ];
+
+            // Return first selector that successfully matches an element in the DOM
+            return selectors.find(selector => $(selector).length > 0)
```

Comment:
Wouldn't it be better to return the first element of the array to avoid getting [0]? WDYT ?

Reply:
`find` does return the first found element :)

If you're referring to this:
```js
        function getAddToCartButton() {
            return $(getAddToCartBtnSelector())[0];
        }
```

I'm using `[0]` _on the result of `$(getAddtoCartBtnSelector())`_ because I want to use the selector as a jQuery selector (which supports `:visible`, whereas `document.querySelector` does not), but need to return a raw `HTMLElement` from the DOM instead of a jQuery object, which the `[]` operator will give me :)